eth/tracers/api.go

```diff
@@ -638,20 +641,28 @@ func (api *API) traceBlock(ctx context.Context, block *types.Block, config *Trac
 			}
 		}()
 	}
+
 	// Feed the transactions into the tracers and return
 	var failed error
 	blockCtx := core.NewEVMBlockContext(block.Header(), api.chainContext(ctx), nil)
+txloop:
 	for i, tx := range txs {
 		// Send the trace task over for execution
-		jobs <- &txTraceTask{statedb: statedb.Copy(), index: i}
+		task := &txTraceTask{statedb: statedb.Copy(), index: i}
+		select {
```

Comment:
Is this more efficient than simply checking `ctx.Err()` at the beginning of the loop?

Reply:
It's not more efficient, I just did it because the send could theoretically block at this point. Canceling the context would unblock it. However, it seems the `jobs` channel is actually buffered, and so the send will always succeed.